bench-vortex/src/bin/clickbench.rs

```diff
@@ -458,49 +465,49 @@ fn benchmark_duckdb_query(
 }
 
 async fn execute_duckdb_query(
-    query_idx: usize,
     query_string: &str,
-    data_path: &std::path::Path,
+    base_url: &Url,
     file_format: Format,
     single_file: bool,
 ) -> anyhow::Result<Duration> {
-    let query_file = tempfile::tempdir()?
-        .path()
-        .join(format!("query_{query_idx}.sql"));
-
-    fs::write(&query_file, query_string)?;
-
     let extension = match file_format {
         Format::Parquet => "parquet",
         Format::OnDiskVortex => "vortex",
         other => vortex_panic!("Format {other} isn't supported on ClickBench"),
     };
 
+    // Base path contains trailing /.
     let file_glob = if single_file {
-        format!(
-            "{}/{extension}/hits.{extension}",
-            data_path.to_string_lossy()
-        )
+        format!("{base_url}{extension}/hits.{extension}")
     } else {
-        format!("{}/{extension}/*.{extension}", data_path.to_string_lossy())
+        format!("{base_url}{extension}/*.{extension}")
     };
 
+    // TODO: double check
+    let file_glob = file_glob.strip_prefix("file://").unwrap();
     let time_instant = Instant::now();
+    let register_tables =
+        format!("CREATE VIEW hits AS SELECT * FROM read_{extension}('{file_glob}')",);
+
+    let repo_root = std::process::Command::new("git")
+        .args(["rev-parse", "--show-toplevel"])
+        .output()
+        .map(|output| String::from_utf8_lossy(&output.stdout).trim().to_string())
+        .unwrap_or_else(|_| ".".to_string());
+
+    let duckdb_path = format!("{repo_root}/duckdb-vortex/build/release/duckdb");
```

Comment:
Not optimal. Maybe we should force symlink the duckdb binary under `/use/local/bin`. Open to suggestions here.

Reply:
Why not optimal? Much much better than leaking outside the repo checkout

Reply:
`repo_root` does not point to the top-level root if the clickbench runner is called whilst the current shell dir is within a submodule.
```
 let repo_root = std::process::Command::new("git")
      .args(["rev-parse", "--show-toplevel"]
```

```
cd vortex/duckdb-vortex/duckdb
git rev-parse --show-toplevel
/Users/lx/code/vortex/duckdb-vortex/duckdb
```

Reply:
How about passing in the DuckDB path explicitly via a flag?

Reply:
> leaking outside the repo checkout

I primarily had CI runners in mind here, and agreed for the general case this doesn't really work.